### PR TITLE
test: write remaining regression Account Settings tests [WPB-25329]

### PR DESCRIPTION
**Pull request by zhannach**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/WPB-25329" title="WPB-25329" target="_blank">TaskWPB-25329</a>  [Web][QA] Write remaining Regression -> Account Settings tests
  </td></table>
  <br />
 

Automated the remaining 3 test case under the Account Settings regression test suite.

Test Cases Automated:
@TC-1938: Automated verification for specific Account Settings edge cases.
@TC-1939: Verified username auto-generation logic across multiple localized character sets and emojis.
@TC-1942: Verified username uniqueness rules, preventing duplicate claims during onboarding and within profile settings.

### CODE REVIEW

File: apps/webapp/test/e2e_tests/specs/AccountSettings/accountSettings.spec.ts [hunk 1]
```diff
@@ -200,6 +200,164 @@ test.describe('account settings', () => {
...
+    await test.step('Validate username uniqueness error during account creating', async () => {
+      await pages.setUsername().setUsername(userA.username);
+      await pages.setUsername().clickNextButton();
+      await expect(pages.setUsername().errorMessage).toContainText('This username is already taken');
+
+      // Proceed successfully with User B's unique username
+      await pages.setUsername().setUsername(userB.username);
+      await pages.setUsername().clickNextButton();
+      await pages.registerSuccess().clickOpenWireWebButton();
+      await modals.confirm().cancelButton.click();
```
Comment (markbrockhoff) [inline apps/webapp/test/e2e_tests/specs/AccountSettings/accountSettings.spec.ts]: The share data consent modal will only be shown after the login, so better give it the extended `LOGIN_TIMEOUT` in case the login takes a bit longer.

```suggestion
      await modals.confirm().cancelButton.click({timeout: LOGIN_TIMEOUT});
```

File: apps/webapp/test/e2e_tests/specs/AccountSettings/accountSettings.spec.ts [hunk 2]
```diff
@@ -200,6 +200,164 @@ test.describe('account settings', () => {
...
+      await expect(pages.registration().submitButton).toBeDisabled();
+
+      await pages.registration().toggleTermsCheckbox();
+      await expect(pages.registration().submitButton).toBeEnabled();
+      await pages.registration().clickSubmitButton();
+      await expect(pages.emailVerification().verificationCodeInputLabel).toBeVisible();
+
+      const verificationCode = await api.brig.getActivationCodeForEmail(userB.email);
+      await pages.emailVerification().enterVerificationCode(verificationCode);
+      await modals.marketingConsent().clickConfirmButton();
```
Comment (markbrockhoff) [inline apps/webapp/test/e2e_tests/specs/AccountSettings/accountSettings.spec.ts:317]: Most of the assertions in here are not related to the test case and don't add value to what the test is trying to verify. E.g. there's no need to re-verify the button is disabled until the checkbox was clicked, other tests already do that.


```suggestion
      await userPageManager.openMainPage();
      await pages.singleSignOn().enterEmailOnSSOPage(userB.email);
      await pages.welcome().clickCreateAccountButton();
      await pages.welcome().clickCreatePersonalAccountButton();

      await pages.registration().fillInUserInfo(userB);
      await pages.registration().toggleTermsCheckbox();
      await pages.registration().clickSubmitButton();
      await expect(pages.emailVerification().verificationCodeInputLabel).toBeVisible();

      const verificationCode = await api.brig.getActivationCodeForEmail(userB.email);
      await pages.emailVerification().enterVerificationCode(verificationCode);
      await modals.marketingConsent().clickConfirmButton();
```

File: apps/webapp/test/e2e_tests/specs/AccountSettings/accountSettings.spec.ts [hunk 3]
```diff
@@ -200,6 +200,164 @@ test.describe('account settings', () => {
...
+    });
+
+    await test.step('Validate username uniqueness error inside Profile Settings', async () => {
+      await components.conversationSidebar().clickPreferencesButton();
+      await expect(pages.account().usernameDisplay).toContainText(userB.username);
+      await pages.account().changeUserName(userA.username);
+      await pages.account().editUserNameButton.click();
+      await expect(userPage.getByText('Already taken')).toBeVisible();
+      await pages.account().userNameInput.press('Enter');
+      await expect(pages.account().usernameDisplay).toContainText(userB.username);
```
Comment (markbrockhoff) [inline apps/webapp/test/e2e_tests/specs/AccountSettings/accountSettings.spec.ts:342]: Please don't tailor the tests to currently broken behavior inside Wire. There's already a bug ticket to fix that the error isn't displayed right away: https://wearezeta.atlassian.net/browse/WPB-25926


```suggestion
      await components.conversationSidebar().clickPreferencesButton();
      await expect(pages.account().usernameDisplay).toContainText(userB.username);
      await pages.account().changeUserName(userA.username);

      // TODO(WPB-25926): Currently the error message isn't shown correctly in the UI
      // await expect(userPage.getByText('Already taken')).toBeVisible();
      // await pages.account().userNameInput.fill(userB.username);
      // await pages.account().userNameInput.press('Enter');

      await expect(pages.account().usernameDisplay).toContainText(userB.username);
```

File: apps/webapp/test/e2e_tests/specs/AccountSettings/accountSettings.spec.ts [hunk 4]
```diff
@@ -200,6 +200,164 @@ test.describe('account settings', () => {
...
+    });
+
+    await test.step('Delete user', async () => {
+      await components.conversationSidebar().clickPreferencesButton();
+      await pages.account().clickDeleteAccountButton();
+      await expect(modals.confirm().modal).toBeVisible();
```
Comment (markbrockhoff) [inline apps/webapp/test/e2e_tests/specs/AccountSettings/accountSettings.spec.ts]: 🎨  Just a suggestion, but what do you think about asserting the modal title instead of only visibility? This check includes the verification for the modal to be visible but makes reading and understanding the test easier :)


```suggestion
      await expect(modals.confirm().modalTitle).toContainText('Delete account');
```

File: apps/webapp/test/e2e_tests/specs/AccountSettings/accountSettings.spec.ts [hunk 5]
```diff
@@ -200,6 +200,164 @@ test.describe('account settings', () => {
...
+
+            await pages.setUsername().clickNextButton();
+            await pages.registerSuccess().clickOpenWireWebButton();
+            await modals.confirm().cancelButton.click();
+
+            await components.conversationSidebar().clickPreferencesButton();
+            await expect(pages.account().usernameDisplay).toContainText(expectedUsername);
+          });
+
+          await test.step('Cleanup: delete account and close page', async () => {
```
Comment (markbrockhoff) [inline apps/webapp/test/e2e_tests/specs/AccountSettings/accountSettings.spec.ts]: Right now the cleanup step would simply be skipped if the test failed along the way. Please make sure it always runs. E.g. by wrapping the step "Verify autogenerated username on registration and profile" in a `try` block and doing the Cleanup step inside its `finally` block.

File: apps/webapp/test/e2e_tests/specs/AccountSettings/accountSettings.spec.ts [hunk 6]
```diff
@@ -200,6 +200,164 @@ test.describe('account settings', () => {
...
+            await modals.marketingConsent().clickConfirmButton();
+          });
+
+          await test.step('Verify autogenerated username on registration and profile', async () => {
+            const defaultUserName = await pages.setUsername().getHandleInputValue();
+            expect(defaultUserName).toContain(expectedUsername);
+
+            await pages.setUsername().clickNextButton();
+            await pages.registerSuccess().clickOpenWireWebButton();
+            await modals.confirm().cancelButton.click();
```
Comment (markbrockhoff) [inline apps/webapp/test/e2e_tests/specs/AccountSettings/accountSettings.spec.ts]: `clickOpenWireWebButton()` initiates a login, let's give it the needed time to prevent flake due to this.

```suggestion

            await expect(modals.confirm().modalTitle).toContainText('share user data');
            await modals.confirm().cancelButton.click({timeout: LOGIN_TIMEOUT});
```

File: apps/webapp/test/e2e_tests/specs/AccountSettings/accountSettings.spec.ts [hunk 7]
```diff
@@ -200,6 +200,164 @@ test.describe('account settings', () => {
...
+            await expect(pages.emailVerification().verificationCodeInputLabel).toBeVisible();
+            const verificationCode = await api.brig.getActivationCodeForEmail(userA.email);
+
+            await pages.emailVerification().enterVerificationCode(verificationCode);
+            await modals.marketingConsent().clickConfirmButton();
+          });
+
+          await test.step('Verify autogenerated username on registration and profile', async () => {
+            const defaultUserName = await pages.setUsername().getHandleInputValue();
+            expect(defaultUserName).toContain(expectedUsername);
```
Comment (markbrockhoff) [inline apps/webapp/test/e2e_tests/specs/AccountSettings/accountSettings.spec.ts]: This breaks playwrights ability to automatically retry the assertion. E.g. if the input is empty for 5ms and the value is taken at that moment it would fail. Better use the assertion `toHaveValue()` for asserting the value of an input to keep retryability.

> Hint: You can pass a regular expression to it in order to check for  a partial match of the value ;)


```suggestion
            await expect(pages.setUsername().handleInput).toHaveValue(new RegExp(expectedUsername));
```

File: apps/webapp/test/e2e_tests/specs/AccountSettings/accountSettings.spec.ts [hunk 8]
```diff
@@ -200,6 +200,164 @@ test.describe('account settings', () => {
...
+        {firstName: 'Jack', expectedUsername: 'jack'},
+        {firstName: 'Jack.Wireson', expectedUsername: 'jackwireson'},
+        {firstName: 'Æéÿüíøšłźçñ', expectedUsername: 'aeeyueioslzcn'},
+        {firstName: 'Даша', expectedUsername: 'dasha'},
+        {firstName: 'داريا', expectedUsername: 'darya'},
+        {firstName: 'Jack😼', expectedUsername: 'jack'},
+      ];
+
+      for (const {firstName, expectedUsername} of usernameScenarios) {
+        await test.step(`Scenario: ${firstName} -> ${expectedUsername}`, async () => {
```
Comment (markbrockhoff) [inline apps/webapp/test/e2e_tests/specs/AccountSettings/accountSettings.spec.ts]: Did you intentionally use a regular for loop instead of running the scenarios in parallel? They are fully independent of each other aren't they? So maybe we could safe some execution time and wouldn't need to increase the test timeout :)

E.g.
```ts
await Promise.all(
  usernameScenarios.map(({firstName, expectedUsername}) => {
    return test.step(`Scenario: ${firstName} -> ${expectedUsername}`, async () => {
      // Existing content for the scenarios
    });
  }),
);
```

File: apps/webapp/test/e2e_tests/pageManager/webapp/pages/account.page.ts [hunk 9]
```diff
@@ -56,8 +58,10 @@ export class AccountPage {
     this.logoutButton = page.getByTestId('do-logout');
     this.editEmailButton = page.getByTestId('go-edit-email');
     this.editDisplayNameButton = page.getByTestId('go-edit-email');
+    this.editUserNameButton = page.getByRole('button', {name: 'Username'});
     this.emailInput = page.getByTestId('enter-email-input');
     this.displayNameInput = page.getByTestId('enter-displayname-input');
+    this.userNameInput = page.getByTestId('enter-username-input');
```
Comment (markbrockhoff) [inline apps/webapp/test/e2e_tests/pageManager/webapp/pages/account.page.ts]: The input is labeled correctly, so let's use the more descriptive `getByLabel()`:


```suggestion
    this.userNameInput = page.getByLabel('Username');
```

**Review by markbrockhoff — CHANGES_REQUESTED** [review]

**Review by markbrockhoff — APPROVED** [review]

**Review by zskhan — APPROVED** [review]
